solver/solve.go

```diff
@@ -174,6 +183,36 @@ func Build(ctx context.Context, c *client.Client, s *session.Session, pw progres
 		statusCh = pw.Status()
 	}
 
+	if info.OutputCapture != nil {
+		captureStatusCh := make(chan *client.SolveStatus)
+		go func(origStatusCh chan *client.SolveStatus) {
+			defer func() {
+				if origStatusCh != nil {
+					close(origStatusCh)
+				}
+			}()
+			for {
+				select {
+				case <-pw.Done():
+					return
+				case <-ctx.Done():
+					return
+				case status, ok := <-captureStatusCh:
+					if !ok {
+						return
+					}
+					for _, log := range status.Logs {
+						info.OutputCapture.Write(log.Data)
```

Comment:
This will capture logs of more than just the run statement. You'll need to isolate the logs based on vertex digest.